config/configparser/config.go

```diff
@@ -533,15 +529,10 @@ func expandEnv(s string) string {
 	})
 }
 
-func unmarshaler(factory component.Factory) func(componentViperSection *config.Parser, intoCfg interface{}) error {
-	if fu, ok := factory.(component.ConfigUnmarshaler); ok {
-		return func(componentParser *config.Parser, intoCfg interface{}) error {
-			return fu.Unmarshal(componentParser.Viper(), intoCfg)
-		}
+func unmarshal(componentSection *config.Parser, intoCfg interface{}) error {
```

Comment:
We should also still look for factory if they implement that and use it, log a deprecation warning and remove the release after.

Reply:
We should still remove the old public interfaces and define here the deprecated interface:

```
type deprecatedUnmarshaler interface {
	Unmarshal(componentViperSection *viper.Viper, intoCfg interface{}) error
}
```

Reply:
We can't move the interface to `configparser` and keep the tests and interface assertions on the component helpers (you get an import cycle). I kept it on component, we can also move it an remove the testing code.

Reply:
I think I misunderstood this the first time, will revert the last commits and just add the interface